### ISSUE TITLE
runc list gives JSON unmarshaling error

### ISSUE THREAD

**Issue by rajasec** [opencontainers/runc#999]
While running the runc list
gave the following error
./runc list
json: cannot unmarshal string into Go value of type uint32

Though /run/runc had state.json for the container.
This came only once. I'll try to reproduce again and report back

**Comment by cyphar** [comment]
One thing I've noticed is that `runc list` is not very resilient to errors while reading files (you don't get _any_ list information if one of the state.json files are bad).

**Comment by crosbymichael** [comment]
this is probably because you have docker containers running and the runc master state changed

**Comment by TrumanLing** [comment]
Agree with crosbymichael that you maybe have containers' directories for ```state.json``` under /run/runc, and those containers were created use old version of config spec.
Actually, configuration field ```net_cls_classid``` has been changed, detailed info is as following:
```bb42f80a libcontainer/configs/cgroup_unix.go (Hushan Jia              2016-07-08 21:07:50 +0800 123) 	NetClsClassid uint32 `json:"net_cls_classid"````

**Comment by rajasec** [comment]
During that problem, there were only 2 containers listed under /run/runc, both had state.json. I switched to master and tried the same. I'm sure that this were not docker containers. As you are mentioning It could be due to old config which might have caused the problem

**Comment by TrumanLing** [comment]
@rajasec 
Would you please try to change the value of field ```net_cls_classid``` form ```"xxx"``` to any number you like, then try to run ```runc list``` and check the output?

**Comment by rajasec** [comment]
Yes you are right, if I change to string, I'm able to see this error back. But I was expecting runc list should give this error only for that respective container,  Error is so generic, but the problem is even though other containers have the right state information, it is just giving one single error.  User is not aware where this error comes from whether container 1 or container 2 ? atleast message to the user this container can help here. I'm working on PR to handle gracefully.

**Comment by cyphar** [comment]
@rajasec is right, we need to make `runc list` more resilient to errors (it will bail on any issue, and not print containers that it can successfully check). My rootless containers patchset somewhat fixes this, and I should probably pull it into a separate PR.
